lib/services/discourse.ts

```diff
@@ -0,0 +1,220 @@
+import * as Promise from 'bluebird';
+import * as _ from 'lodash';
+import * as path from 'path';
+import * as request from 'request-promise';
+import {
+    MessageEmitResponse,
+    MessageEvent,
+    MessageWorkerEvent,
+    ReceiptContext,
+} from '../utils/message-types';
+import {
+    DiscourseMessageEmitContext,
+    DiscoursePost,
+} from './discourse-types';
+import {
+    MessageService,
+} from './message-service';
+import {
+    ServiceEmitter,
+    ServiceListener,
+} from './service-types';
+
+/**
+ * Class for interacting with the discourse API
+ * Is a MessageService, ServiceListener and ServiceEmitter
+ */
+export class DiscourseService extends MessageService implements ServiceListener, ServiceEmitter {
+    private static _serviceName = path.basename(__filename.split('.')[0]);
+    private topicCache = new Map<string, any>();
+
+    /**
+     * Retrieve the comments in a thread that match an optional filter
+     * @param event details to identify the event
+     * @param filter regex of comments to match
+     */
+    public fetchThread(event: ReceiptContext, filter: RegExp): Promise<string[]> {
+        // Check that the event being asked about orginated with us
+        if(event.source !== this.serviceName) {
+            return Promise.reject(new Error('Cannot get discourse thread from non-discourse event'));
```

Comment:
Yes, I misread this, it's not in a Promise, so that won't end up in the chain. :+1: